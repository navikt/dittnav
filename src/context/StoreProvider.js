import React, { createContext, useReducer } from 'react';
import { node } from 'prop-types';
import storeReducer, { initialStoreState } from '../reducers/storeReducer';

export const StoreContext = createContext(null);

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(storeReducer, initialStoreState());

  const addBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const addInaktiveBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_INAKTIVE_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const removeBeskjed = (beskjed) => dispatch({
    type: 'REMOVE_BESKJED',
    payload: beskjed,
  });

  const addInaktivBeskjed = (beskjed) => dispatch({
    type: 'ADD_INAKTIV_BESKJED',
    payload: beskjed,
  });

  const toggleInnloggingsModal = () => dispatch({
    type: 'TOGGLE_INNLOGGINGSMODAL',
  });

  const setError = () => dispatch({
    type: 'SET_ERROR',
  });

  return (
    <StoreContext.Provider value={{
      state,
      addBeskjeder,
      addInaktivBeskjed,
      addInaktiveBeskjeder,
      removeBeskjed,
      toggleInnloggingsModal,
      setError,
    }}
    >
      {children}
    </StoreContext.Provider>
  );
};

StoreProvider.propTypes = {
  children: node.isRequired,
};

export default StoreProvider;
