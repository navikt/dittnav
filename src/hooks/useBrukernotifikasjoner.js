import { useQuery } from 'react-query';
import { fetchBeskjeder, fetchOppgaver, fetchInnbokser } from '../Api';
import useStore from './useStore';

const useBeskjeder = () => {
  const { state, addBeskjeder, setError } = useStore();

  useQuery('beskjeder', fetchBeskjeder, {
    onSuccess: addBeskjeder,
    onError: setError,
  });

  return state.beskjeder;
};

const useOppgaver = () => (
  useQuery('oppgaver', fetchOppgaver, { onError: useStore().setError })
);

const useInnbokser = () => (
  useQuery('innbokser', fetchInnbokser, { onError: useStore().setError })
);

const useBrukernotifikasjoner = () => (
  [useBeskjeder(), useOppgaver(), useInnbokser()]
);

export default useBrukernotifikasjoner;
