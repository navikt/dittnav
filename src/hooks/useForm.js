import { useState } from 'react';

const useForm = (initialState, sendSubmit) => {
  const [state, setState] = useState(initialState);

  const handleChange = (event) => {
    event.persist();
    setState(st => ({
      ...st,
      [event.target.name]: event.target.value,
    }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    sendSubmit(state);
  };

  return [state, handleChange, handleSubmit];
};

export default useForm;
