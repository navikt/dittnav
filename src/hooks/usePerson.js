import { useQuery } from 'react-query';
import { fetchOppfolging, fetchMeldekort, fetchMeldinger, fetchPersonIdent, fetchPersonNavn } from '../Api';
import useStore from './useStore';

export const useOppfolging = () => (
  [useQuery('oppfolging', fetchOppfolging, { onError: useStore().setError })]
);

export const useMeldekort = () => (
  [useQuery('meldekort', fetchMeldekort, { onError: useStore().setError })]
);

export const useMeldinger = () => (
  [useQuery('meldinger', fetchMeldinger, { onError: useStore().setError })]
);

export const useIdent = (enabled) => (
  [useQuery('ident', fetchPersonIdent, { enabled, onError: useStore().setError })]
);

export const useNavn = () => (
  [useQuery('navn', fetchPersonNavn, { onError: useStore().setError })]
);

const usePerson = () => (
  [useOppfolging(), useMeldekort(), useMeldinger(), useNavn()]
);

export default usePerson;
