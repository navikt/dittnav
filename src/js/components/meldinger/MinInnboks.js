import React, { Component } from 'react';
import PropTypes from 'prop-types';
import i18n from 'translations/i18n';
import { FormattedMessage as F, injectIntl, intlShape } from 'react-intl';

const formatFlereEn = (length, i18String) => `${i18String}${length === 1 ? 'en' : 'flere'}`;

const getMessagesIcon = (type) => {
  switch (type) {
    case 'DOKUMENT_VARSEL':
      return 'document-icon';
    case 'OPPGAVE_VARSEL':
      return 'registration-icon';
    default:
      return 'mininnboks-default-icon';
  }
};

class MinInnboks extends Component {
  render() {
    const { numberToWord } = i18n[this.props.intl.locale];
    const messages = this.props.mininnboks;
    return (
      <React.Fragment>
        {messages && messages.map(message => (
          <a
            key={message.type}
            data-ga={`Dittnav/Varsel/${message.type.toLowerCase()} melding`}
            className="message clickable"
            href={message.url}
          >
            <span className={`icon ${getMessagesIcon(message.type)}`} aria-label={`${message.type.toLowerCase().replace(/_/g, ' ')} ikon`} />
            <div className="texts">
              <p><F id={formatFlereEn(message.antall, `mininnboks.${message.type.toLowerCase()}.meldinger.`)} values={{ count: numberToWord(message.antall) }} /></p>
            </div>
          </a>
        ))}
      </React.Fragment>
    );
  }
}

export const MinInnboksType = PropTypes.arrayOf(PropTypes.shape({ type: PropTypes.string.isRequired, uri: PropTypes.string }));

MinInnboks.propTypes = {
  mininnboks: MinInnboksType,
  intl: intlShape.isRequired, // eslint-disable-line react/no-typos
};

MinInnboks.defaultProps = {
  mininnboks: [],
};

export default injectIntl(MinInnboks);
