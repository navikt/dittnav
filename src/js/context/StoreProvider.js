import React, { createContext, useReducer } from 'react';
import { node } from 'prop-types';
import storeReducer, { initialStoreState } from '../reducers/storeReducer';

export const StoreContext = createContext(null);

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(storeReducer, initialStoreState());

  const addBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const addInaktiveBeskjeder = (nyeBeskjeder) => dispatch({
    type: 'ADD_INAKTIVE_BESKJEDER',
    payload: nyeBeskjeder,
  });

  const removeBeskjed = (beskjed) => dispatch({
    type: 'REMOVE_BESKJED',
    payload: beskjed,
  });

  const addInaktivBeskjed = (beskjed) => dispatch({
    type: 'ADD_INAKTIV_BESKJED',
    payload: beskjed,
  });

  const toggleInnloggingsModal = () => dispatch({
    type: 'TOGGLE_INNLOGGINGSMODAL',
  });

  const addOppgaver = (oppgaver) => dispatch({
    type: 'ADD_OPPGAVER',
    payload: oppgaver,
  });

  const addInnbokser = (innbokser) => dispatch({
    type: 'ADD_INNBOKSER',
    payload: innbokser,
  });

  const addInaktiveOppgaver = (inaktiveOppgaver) => dispatch({
    type: 'ADD_INAKTIVE_OPPGAVER',
    payload: inaktiveOppgaver,
  });

  const addInaktiveInnbokser = (inaktiveInnbokser) => dispatch({
    type: 'ADD_INAKTIVE_INNBOKSER',
    payload: inaktiveInnbokser,
  });

  const addInnloggingsstatus = (innloggingsstatus) => dispatch({
    type: 'ADD_INNLOGGINGSSTATUS',
    payload: innloggingsstatus,
  });

  const addOppfolging = (oppfolging) => dispatch({
    type: 'ADD_OPPFOLGING',
    payload: oppfolging,
  });

  const addMeldekort = (meldekort) => dispatch({
    type: 'ADD_MELDEKORT',
    payload: meldekort,
  });

  const addNavn = (navn) => dispatch({
    type: 'ADD_NAVN',
    payload: navn,
  });

  const addIdent = (Ident) => dispatch({
    type: 'ADD_IDENT',
    payload: Ident,
  });

  const addPaabegynteSoknader = (paabegynteSoknader) => dispatch({
    type: 'ADD_PAABEGYNTESOKNADER',
    payload: paabegynteSoknader,
  });

  const addMeldinger = (meldinger) => dispatch({
    type: 'ADD_MELDINGER',
    payload: meldinger,
  });

  const addSakstema = (sakstema) => dispatch({
    type: 'ADD_SAKSTEMA',
    payload: sakstema,
  });

  const setOppfolgingFailed = () => dispatch({
    type: 'OPPFOLGING_FAILED',
  });

  const setError = () => dispatch({
    type: 'ERROR',
  });

  return (
    <StoreContext.Provider value={{
      state,
      addBeskjeder,
      addOppgaver,
      addInnbokser,
      addInaktivBeskjed,
      addInaktiveBeskjeder,
      addInnloggingsstatus,
      addInaktiveOppgaver,
      addInaktiveInnbokser,
      addOppfolging,
      addMeldekort,
      addNavn,
      addIdent,
      addPaabegynteSoknader,
      addMeldinger,
      addSakstema,
      removeBeskjed,
      toggleInnloggingsModal,
      setOppfolgingFailed,
      setError,
    }}
    >
      {children}
    </StoreContext.Provider>
  );
};

StoreProvider.propTypes = {
  children: node.isRequired,
};

export default StoreProvider;
