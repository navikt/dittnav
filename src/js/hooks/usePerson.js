import { useQuery } from 'react-query';
import Api from '../Api';

export const useOppfolging = () => (
  [useQuery('oppfolging', Api.fetchOppfolging)]
);

export const useMeldekort = () => (
  [useQuery('meldekort', Api.fetchMeldekort)]
);

export const useMeldinger = () => (
  [useQuery('meldinger', Api.fetchMeldinger)]
);

export const useNavn = () => (
  [useQuery('navn', Api.fetchPersonNavn)]
);

export const useIdent = () => (
  [useQuery('ident', Api.fetchPersonIdent)]
);
