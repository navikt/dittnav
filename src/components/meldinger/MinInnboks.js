import React from 'react';
import { FormattedMessage as F, useIntl } from 'react-intl';
import i18n from '../../language/i18n';
import LenkepanelMedIkon from '../common/LenkepanelMedIkon';
import PanelOverskrift from '../common/PanelOverskrift';
import IkonOppgave from '../../assets/IkonOppgave';
import IkonMinInnboks from '../../assets/IkonMinInnboks';
import { GoogleAnalyticsCategory } from '../../utils/googleAnalytics';
import { useMeldinger } from '../../hooks/usePerson';
import IkonBrevVedtak from '../../assets/IkonBrevVedtak';

const getMinInnboksIcon = (type) => {
  switch (type) {
    case 'DOKUMENT_VARSEL':
      return <IkonBrevVedtak />;
    case 'OPPGAVE_VARSEL':
      return <IkonOppgave />;
    default:
      return <IkonMinInnboks />;
  }
};

const isDokumentVarsel = (type) => (
  type === 'DOKUMENT_VARSEL'
);

const createOverskrift = (message, numberToWord, formatFlereEn) => {
  const overskrift = (
    <F id={formatFlereEn(message.antall, `mininnboks.${message.type.toLowerCase()}.meldinger.`)} values={{ count: numberToWord(message.antall) }} />
  );

  return (
    <PanelOverskrift overskrift={overskrift} type="Element" />
  );
};

const MinInnboks = () => {
  const [{ data: meldinger, isSuccess }] = useMeldinger();
  const { numberToWord } = i18n[useIntl().locale];
  const formatFlereEn = (length, i18String) => `${i18String}${length === 1 ? 'en' : 'flere'}`;

  if (!isSuccess) {
    return null;
  }

  return (
    <>
      {meldinger.content.map(message => (
        <LenkepanelMedIkon
          key={message.type}
          className={isDokumentVarsel(message.type) ? 'infomelding beskjed' : 'infomelding innboks'}
          alt="Melding fra mininnboks"
          overskrift={createOverskrift(message, numberToWord, formatFlereEn)}
          href={message.url}
          gaCategory={GoogleAnalyticsCategory.Forside}
          gaAction={`${message.type.toLowerCase()}`}
        >
          {getMinInnboksIcon(message.type)}
        </LenkepanelMedIkon>
      ))}
    </>
  );
};

export default MinInnboks;
