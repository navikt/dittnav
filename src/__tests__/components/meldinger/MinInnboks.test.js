import * as React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import MinInnboks from 'components/meldinger/MinInnboks'
import wrapIntl from 'utils/intlTestHelper';
import sinon from 'sinon';
import StoreProvider from '../../../context/StoreProvider';
import * as usePerson from '../../../hooks/usePerson';

jest.mock('react-ga');
let sandbox = null;

beforeEach(() => {
  sandbox = sinon.createSandbox();
});

afterEach(() => {
  usePerson.useMeldinger.restore();
});

test('MinInnboks with one document', () => {
  const mininnboks = [{"type": "DOKUMENT_VARSEL", "antall": 1, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));
  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with two documents', () => {
  const mininnboks = [
    {"type": "DOKUMENT_VARSEL", "antall": 2, "url": "https://tjenester-t6.nav.no/mininnboks"}
  ];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));
  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with two documents but different types', () => {
  const mininnboks = [
    {"type": "DOKUMENT_VARSEL", "antall": "1", "url": "https://tjenester-t6.nav.no/mininnboks"},
    {"type": "UBESVART", "antall": "1", "url": "https://tjenester-t6.nav.no/someother"}
  ];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));
  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with one task', () => {
  const mininnboks = [{"type": "OPPGAVE_VARSEL", "antall": 1, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with two tasks', () => {
  const mininnboks = [{"type": "OPPGAVE_VARSEL", "antall": 1, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with zero tasks', () => {
  const mininnboks = [{"type": "OPPGAVE_VARSEL", "antall": 0, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with one unread task and one unread', () => {
  const mininnboks = [
    {"type": "OPPGAVE_VARSEL", "antall": 1, "url": "https://tjenester-t6.nav.no/mininnboks"},
    {"type": "ULEST", "antall": 1, "url": "https://tjenester-t6.nav.no/mininnboks"}
  ];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks mininnboks={mininnboks} />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with two unread', () => {
  const mininnboks = [{"type": "ULEST", "antall": 2, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with one unanswered', () => {
  const mininnboks = [{"type": "UBESVART", "antall": 1, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with two unanswered', () => {
  const mininnboks = [{"type": "UBESVART", "antall": 2, "url": "https://tjenester-t6.nav.no/mininnboks"}];

  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: mininnboks },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
    ));

  expect(component.toJSON()).toMatchSnapshot();
});

test('MinInnboks with empty input', () => {
  sandbox.stub(usePerson, 'useMeldinger')
    .returns([{
      data: { content: [] },
      isLoading: false,
      isSuccess: true,
    }]);

  const component = ReactTestRenderer.create(wrapIntl(
    <StoreProvider>
      <MinInnboks />
    </StoreProvider>
  ));

  expect(component.toJSON()).toMatchSnapshot();
});
