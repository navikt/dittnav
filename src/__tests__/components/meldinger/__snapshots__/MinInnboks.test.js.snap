// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinInnboks with empty input 1`] = `null`;

exports[`MinInnboks with null input 1`] = `null`;

exports[`MinInnboks with one document 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/dokument_varsel melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="dokument varsel ikon"
    className="icon document-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har ett ulest dokument i din innboks
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with one task 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/oppgave_varsel melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="oppgave varsel ikon"
    className="icon registration-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har én ulest oppgave
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with one unanswered 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/ubesvart melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="ubesvart ikon"
    className="icon mininnboks-default-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har ett ubesvart spørsmål i din innboks
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with one unread task and one unread 1`] = `
Array [
  <a
    className="message clickable"
    data-ga="Dittnav/Varsel/oppgave_varsel melding"
    href="https://tjenester-t6.nav.no/mininnboks"
  >
    <span
      aria-label="oppgave varsel ikon"
      className="icon registration-icon"
    />
    <div
      className="texts"
    >
      <p>
        <span>
          Du har én ulest oppgave
        </span>
      </p>
    </div>
  </a>,
  <a
    className="message clickable"
    data-ga="Dittnav/Varsel/ulest melding"
    href="https://tjenester-t6.nav.no/mininnboks"
  >
    <span
      aria-label="ulest ikon"
      className="icon mininnboks-default-icon"
    />
    <div
      className="texts"
    >
      <p>
        <span>
          Du har én ulest melding i din innboks
        </span>
      </p>
    </div>
  </a>,
]
`;

exports[`MinInnboks with two documents 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/dokument_varsel melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="dokument varsel ikon"
    className="icon document-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har to uleste dokumenter i din innboks
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with two documents but different types 1`] = `
Array [
  <a
    className="message clickable"
    data-ga="Dittnav/Varsel/dokument_varsel melding"
    href="https://tjenester-t6.nav.no/mininnboks"
  >
    <span
      aria-label="dokument varsel ikon"
      className="icon document-icon"
    />
    <div
      className="texts"
    >
      <p>
        <span>
          Du har  uleste dokumenter i din innboks
        </span>
      </p>
    </div>
  </a>,
  <a
    className="message clickable"
    data-ga="Dittnav/Varsel/ubesvart melding"
    href="https://tjenester-t6.nav.no/someother"
  >
    <span
      aria-label="ubesvart ikon"
      className="icon mininnboks-default-icon"
    />
    <div
      className="texts"
    >
      <p>
        <span>
          Du har  ubesvarte spørsmål i din innboks
        </span>
      </p>
    </div>
  </a>,
]
`;

exports[`MinInnboks with two tasks 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/oppgave_varsel melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="oppgave varsel ikon"
    className="icon registration-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har én ulest oppgave
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with two unanswered 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/ubesvart melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="ubesvart ikon"
    className="icon mininnboks-default-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har to ubesvarte spørsmål i din innboks
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with two unread 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/ulest melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="ulest ikon"
    className="icon mininnboks-default-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har to uleste meldinger i din innboks
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks with zero tasks 1`] = `
<a
  className="message clickable"
  data-ga="Dittnav/Varsel/oppgave_varsel melding"
  href="https://tjenester-t6.nav.no/mininnboks"
>
  <span
    aria-label="oppgave varsel ikon"
    className="icon registration-icon"
  />
  <div
    className="texts"
  >
    <p>
      <span>
        Du har  uleste oppgaver
      </span>
    </p>
  </div>
</a>
`;

exports[`MinInnboks without any messages 1`] = `null`;
